src/base/RegexPattern.h

```diff
@@ -22,26 +23,42 @@ class RegexPattern
 
 public:
     RegexPattern() = delete;
-    RegexPattern(int aFlags, const char *aPattern);
+    RegexPattern(const SBuf &aPattern, int aFlags);
     ~RegexPattern();
 
-    // regex type varies by library, usually not safe to copy
-    RegexPattern(const RegexPattern &) = delete;
-    RegexPattern &operator =(const RegexPattern &) = delete;
+    RegexPattern(RegexPattern &&) = delete; // no copying of any kind
```

Comment:
I removed these unused (and slightly problematic) RegexPattern move optimizations to simplify. Technically, calling POSIX regfree() with a zeroed regex was a UB because regfree() is only defined for compiled regexes. Since these move optimizations were actually unused (and are not expected to be used), we should just delete this (slightly) problematic code path.